src/js/utils/element-factory.js

```diff
@@ -70,9 +70,19 @@ export const getInput = (type, label, value = "", describedBy, name) => {
   return $input;
 };
 
+export const getFieldset = () => {
+  return $("<fieldset />");
+};
+
+export const getLegend = (label) => {
+  return $("<legend />", {
+    text: label,
+  });
+};
+
 export const getLabel = (label) => {
   return $("<label />", {
-    for: label,
+    for: label.split(" ").join("-").toLocaleLowerCase(),
```

Comment:
Please don't munge between data and human-readable versions - this is hiding the fact that something's not using the interface correctly and is just going to break things invisibly.

Reply:
Let's make the code be a bit stricter rather than spiral down more and more accidentally-working or silently failing code that will have to be fixed eventually anyway

Reply:
Now that Matt added `label` elements to Webflow, I can probably replace all uses of this with the Webflow version. I am slowly moving away from using the element factory. It was always meant to only be a stop gap.